# Medidas de Dispersión

In [3]:
import pandas as pd

## Objetivos:

Aplicar medidas de dispersión (rango, desviación estándar y coeficiente de variación) en el análisis de bases de datos, utilizando Python como herramienta de apoyo.

*Nota: Este laboratorio está diseñado para desarrollar razonamiento estadístico. Evita el uso de herramientas de IA para responder las preguntas a menos que esto sea explícitamente sugerido* 🤖. *En su lugar, enfócate en aplicar los conceptos revisados.*

## Actividad 1

El gobierno de la ciudad de San Francisco ha publicado los sueldos base de sus funcionarios públicos entre los años 2011 y 2014. Esta base incluye trabajadores con distintos cargos y niveles de responsabilidad.
Como analista de datos, tu tarea es describir y cuantificar la variabilidad de los sueldos base usando medidas estadísticas de dispersión.

**Utiliza la siguiente base de datos para responder las actividades 1, 2 y 3.**

Descarga la base de datos **<a href="https://drive.google.com/uc?export=download&id=1Ry63iApbDJaY_C1HfR5s-kIX5ldWcGW8
">aquí</a>**.

*Observación: Considera que la variable Pago base está expresada en dólares estadounidenses (USD) y corresponde a ingresos anuales.*



In [4]:
df_1 = pd.read_csv("Actividad_1_2_3.csv")
df_1

Id  Nombre del empleado  \
0      28822          SUSAN KYAUK   
1      29857           JOYCE WONG   
2      30163        OSCAR JIMENEZ   
3      30871          FATIMA AZAM   
4      31788        PAULINE PEREA   
...      ...                  ...   
2265  148382      Ariana N Campos   
2266  148383       Gustavo Serina   
2267  148384        Arturo Durazo   
2268  148385  Patrick J O'Donnell   
2269  148386      Phoebe E Barkan   

                                     Título profesional  Pago base   Año  
0                    Supervisor De Cuentas De Pacientes      55795  2011  
1                                    Auxiliar De Nómina      20860  2011  
2                              Supervisor De Recreación     116820  2011  
3                                               Portero      26265  2011  
4                                              Custodio      36850  2011  
...                                                 ...        ...   ...  
2265                        Becario De Servicio Público      55144  2014  
2266  Gerente De Comercio De La Junta, Grupo 2, M = ...     112161  2014  
2267                              Trabajador De Salud 3     117360  2014  
2268                                Líder De Recreación      90979  2014  
2269                                Líder De Recreación      46979  2014  

[2270 rows x 5 columns]

1. Identifica la población y muestra del estudio.

Pobalción: Funcionarios públicos entre los años 2011 y 2014 de la ciudad de San Francisco. 
Muestra: 2270 Funcionarios públicos entre los años 2011 y 2014 de la ciudad de San Francisco.

2. Muestra las tres primeras y tres últimas filas de la base de datos.

3. Calcula el rango de la variable *Pago base*, ¿este valor representa la realidad de todos los trabajadores?

In [5]:
#rango = MAX - MIN
min_sueldo = df_1['Pago base'].min()
max_sueldo = df_1['Pago base'].max()
rango_sueldo = max_sueldo - min_sueldo
print(f"El rango de sueldo de los empleados públicos de la ciudad de San Francisco es: ${rango_sueldo} USD.")

El rango de sueldo de los empleados públicos de la ciudad de San Francisco es: $272497 USD.


Respuesta: El rango no representa la realidad de todos los trabajadores, ya que solo considera los valores extremos y el rango calculado es muy amplio.

4. Calcula la media, mediana, desviación estándar y coeficiente de variación del sueldo base. ¿La dispersión es alta o baja respecto al promedio? ¿Qué medida (media o mediana) consideras más representativa para describir el sueldo típico? Justifica tu respuesta.



In [6]:
media = df_1['Pago base'].mean()
mediana = df_1['Pago base'].median()
desviacion = df_1['Pago base'].std()
cv = desviacion/media * 100
print(f"La media es de: ${media:.2f} USD. \nLa mediana es de: ${mediana} USD. \nLa desviación es: ${desviacion:.2f} USD. \nEl coeficiente de variación es de: {cv:.1f} %")

La media es de: $77087.22 USD. 
La mediana es de: $74711.5 USD. 
La desviación es: $31121.66 USD. 
El coeficiente de variación es de: 40.4 %


## Actividad 2

Un equipo de análisis de datos quiere comparar la variabilidad de los sueldos base entre distintos años para evaluar si ha habido más o menos estabilidad en las remuneraciones del empleo público.

Trabajarás con la base de datos que ha publicado el gobierno de San Francisco con información de los sueldos base de sus funcionarios públicos entre los años 2011 y 2014. Esta base incluye trabajadores con distintos cargos y niveles de responsabilidad.

**Utiliza la siguiente base de datos para responder las actividades 1, 2 y 3.**

Descarga la base de datos **<a href="https://drive.google.com/uc?export=download&id=1Ry63iApbDJaY_C1HfR5s-kIX5ldWcGW8
">aquí</a>**.

*Observación: Considera que la variable Pago base está expresada en dólares estadounidenses (USD) y corresponde a ingresos anuales.*

1. Importa la base de datos y crea una tabla que agrupe los valores de *Pago base*. Luego, calcula la media, la desviación estándar y el coeficiente de variación por año.

In [7]:
agrupa_anio = df_1.groupby('Año')['Pago base'].agg(['mean', 'std'])
agrupa_anio['Coeficiente de Variacon %'] = (agrupa_anio['std']/agrupa_anio['mean'])*100
round(agrupa_anio, 1)

mean      std  Coeficiente de Variacon %
Año                                              
2011  79206.6  26326.9                       33.2
2012  74188.2  33407.9                       45.0
2013  76963.0  32361.8                       42.0
2014  89173.0  42143.0                       47.3

2. ¿En cuál año se observa mayor dispersión absoluta en los sueldos base? Justifica usando la desviación estándar.

Se observa el mayor la mayor dispersión absoluta el año 2014, ya que es el valor más alto.

3. ¿Cuál año tiene mayor heterogeneidad? Justifica usando el coeficiente de variación.

El año más heterogéneo es el año 2014, porque su CV es el más alto o más lejano del 0%.

4. ¿Qué ventajas tiene el coeficiente de variación frente a la desviación estándar al realizar comparaciones?

5. ¿Qué interpretación puedes hacer al comparar los datos del año 2011 con los del 2014?

## Actividad 3

Un equipo de análisis de datos busca estudiar qué tan parecidos o distintos son los sueldos base entre trabajadores que ejercen el mismo cargo. Para ello, se analizarán los cinco títulos profesionales más frecuentes en la base de datos publicada por el gobierno de San Francisco, la cual contiene información de los sueldos base de funcionarios públicos entre los años 2011 y 2014. Esta base incluye trabajadores con distintos niveles de responsabilidad.

Como analista, tu tarea es calcular y comparar medidas estadísticas de dispersión dentro de cada cargo, para evaluar la variabilidad de las remuneraciones en estos grupos.

**Utiliza la siguiente base de datos para responder las actividades 1, 2 y 3.**

Descarga la base de datos **<a href="https://drive.google.com/uc?export=download&id=1Ry63iApbDJaY_C1HfR5s-kIX5ldWcGW8
">aquí</a>**.

*Observación: Considera que la variable Pago base está expresada en dólares estadounidenses (USD) y corresponde a ingresos anuales.*

1. Crea una tabla de frecuencias absolutas que muestre los cuatro cargos  más frecuentes de San Francisco, usando la variable *Título profesional*.

2. ¿Qué observas respecto a los cargos más comunes? ¿Te sorprende alguno?

3. De acuerdo con la tabla realizada en el inciso anterior, crea un DataFrame que contenga solo los cuatro cargos con mayor presencia en la ciudad de San Francisco, para esto debes filtrar tu DataFrame original. Luego, muestra las cinco primeras filas y las cinco últimas filas.

$\hspace{1cm}$ *Para filtrar rápidamente puedes almacenar los cargos frecuentes en una variable y luego utilizar la función isin.*

4. Utiliza el DataFrame filtrado del inciso anterior para crear una tabla con el cálculo de la media, mediana, desviación estándar y coeficiente de variación del sueldo base (*Pago base*).

5. ¿Cuál de los cargos tiene sueldos más homogéneos? Justifica.